README.md

```diff
@@ -3,33 +3,161 @@
+
+Target files are created by rendering a corresponding template which refers its configuration via the `@configs` hash.
+
+You can configure PDK to make files executable if the file is executable in the template. To do this, include `manage_execute_permissions: true` either in the target file or globally via the common config key.
 
 ## Basic Usage
 
-Templates like this one can be used in conjunction with the PDK. As default the PDK itself uses the templates within this repository to render files for use within a module. Templates can be passed to the PDK as a flag for several of the commands.
+By default PDK uses the templates within this repository to render files for use within a module. To use a third-party template, pass its URL using the `–-template-url` flag as shown in the following example: 
 
 ```bash
 pdk convert --template-url https://github.com/puppetlabs/pdk-templates
 ```
 
-Please note that the template only needs to be passed in once if you wish to change it, every command run on the PDK will use the last specified template.
-For more on basic usage and more detailed description of the PDK in action please refer to the [PDK documentation](https://github.com/puppetlabs/pdk/blob/main/README.md).
+You can also specify branches by using the `--template-ref`, like this:
+
+```bash
+pdk convert --template-url https://github.com/puppetlabs/pdk-templates --template-ref branch_name
+```
+
+Note: Commands run on PDK will use the last specified template repository. 
+
+When you have specified the template repository, you can get started with creating a module, if you don't already have one. Once your module is created, you can customize PDK-templates by editing the `.sync.yml` to override the default configuration, and then running `PDK update` on the module so that the new settings are applied.
+
+For more information on basic usage and more detailed description of PDK in action please refer to [PDK documentation](https://github.com/puppetlabs/pdk/blob/main/README.md), where you can find detailed instructions on how to create, convert and update modules, as well as other useful commands.
+
+## Adjusting templates via /sync.yml
+
+> While we provide a basic template it is likely that it will not match what you need exactly, as such we allow it to be altered or added to through the use of the `.sync.yml` file.
+
+Here we will provide some examples of how you can use `.sync.yml`. For a more comprehensive list of default settings, check the [PDK default config values](#pdk-default-config-values) section. 
+
+### Adding configuration values
+
+Values can be added to the data passed to the templates, thus allowing you to make changes such as testing against additional operating systems or adding new rubocop rules.
+
+To add a value to an array simply place it into the `.sync.yml` file as shown below, here is the structure to add an additional unit test run against Puppet 4:
```

Comment:
```suggestion
To add a value to an array simply place it into the `.sync.yml` file as shown below, here is the structure to add an additional unit test run against Puppet 6:
```
To match the new example.
Could we just set to `~> 8.0` here to keep it up to date?
`rvm` could be updated to 3.2.0 just to ensure its a correct example as well

Reply:
Will amend the message and bump rvm. But I believe the point of the example was to showcase adding new unit test cases against something that was not likely to be there. For example, unsupported puppet versions. thats why it wasnt updated to begin with. I could be misinterpreting the original documentation tho.